apps/dex/src/pages/swap.tsx

```diff
@@ -370,25 +372,36 @@ const SwapPage = () => {
     toDenom,
   ]);
 
-  const swapButtonMsg = (() => {
+  const validationError = useMemo(() => {
     if (signerStatus === "resolved" && signer === undefined) {
-      return "Please Connect Sif Wallet";
+      return new Error("Please Connect Sif Wallet");
     }
 
-    if (!isReady) {
-      return "Loading";
+    if (fromAmountDecimal?.toFloatApproximation() === 0) {
+      return new Error();
```

Comment:
The error here is simply a wrapper for the error message, which then would be used to display text in the UI like so `{validationError.message}`.

Reply:
Also not sure if a constant errors file would have much value since this is simply a form validation error. You would saturate the centralised error constants really quickly with things like: `Wallet not connected`, `Invalid value`, `Can't be negative`, `Required field missing`, etc

Reply:
You can have one for Swap, Margin, Pools, etc... the main `SifchainUIError` object is shared, but the messages are specific to the individual parts of the app... imagine if we isolated each page in its own app, we could easily scan all errors that we are returning/validating.

We don't need to implement that in this PR. It can be a concept we can follow in general. Having random string-only errors is hard to track after we aggregate errors. We should have a namespace for each part of the app too.

From an observability perspective to search and aggregate errors. Imagine we have `<namespace>.100` for form validation errors, and each form in the app uses it (e.g. `SWAP.100.WALLET_NOT_CONNECTED`, `MARGIN.100.MISSING_LEVERAGE`). 

We can query "select all *.100 from errors_frontend" to understand where users are having issues with forms (perhaps we need a better form layout or help text etc)

Reply:
Well in my head the ideal solution would be to use this really: https://react-hook-form.com/get-started#Handleerrors simple, straight to the point

Seen too many horror stories with custom error types n constants, look no further actually :))
https://github.com/Sifchain/sifchain-ui/search?q=error

Reply:
For actual error yeh, make sense, but for input validation, things could get out of hand/overengineered really quick

Reply:
If you don't have a goal for implementation, everything is a horror story 😅

Since we are not tracking any errors, we can ignore this conversation. 😂 

And to be honest, adding tracking for form validation can be unnecessary, it all depends on the product in the end of the day 🤷‍♂️